Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Models.CCT.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Services.RecIM
+{
+    public class ActivityService : IActivityService
+    {
+        private readonly CCTContext _context;
+        private readonly ISeriesService _seriesService;
+
+        public ActivityService(CCTContext context, ISeriesService seriesService)
+        { 
+            _context = context;
+            _seriesService = seriesService;
+        }
+
+
+        public IEnumerable<ActivityViewModel> GetActivities()
+        {
+            var activities = _context.Activity
+                            .Select(a => new ActivityViewModel
+                            {
+                                ID = a.ID,
+                                Name = a.Name,
+                                RegistrationStart = a.RegistrationStart,
+                                RegistrationEnd = a.RegistrationEnd,
+                                Sport = _context.Sport
+                                        .FirstOrDefault(s => s.ID == a.SportID),
+                                Status = _context.ActivityStatus
+                                        .FirstOrDefault(s => s.ID == a.StatusID)
+                                        .Description,
+                                MinCapacity = a.MinCapacity,
+                                MaxCapacity = a.MaxCapacity,
+                                SoloRegistration = a.SoloRegistration,
+                                Logo = a.Logo,
+                                Completed = a.Completed,
+                                Series = a.Series
+                                        .Select(s => new SeriesViewModel
+                                        {
+                                            ID = s.ID,
+                                            Name = s.Name,
+                                            StartDate = s.StartDate,
+                                            EndDate = s.EndDate,
+                                            Type = _context.SeriesType
+                                                    .FirstOrDefault(st => st.ID == s.TypeID)
+                                                    .Description
+                                        }).ToList()
+                            });
```

Comment:
It would be better to use an implicit operator on the RecIM ActivityViewModel for this conversion.

Reply:
Okay, perhaps implicit operators are not the right route here. At the very least, I think this should be factored out to a helper method.